File: 1720. Decode XORed Array/solution.js
/**
 * @param {number[]} encoded
 * @param {number} first
 * @return {number[]}
 */
var decode = function(encoded, first) {
    let ans = [first];
    for (let i of encoded) {
        ans.push(ans[ans.length - 1] ^ i);
    }
    return ans;
};

File: 1720. Decode XORed Array/solution.cpp
class Solution {
public:
    vector<int> decode(vector<int>& encoded, int first) {
        vector<int>ans;
        ans.push_back(first);
        for(auto i: encoded){
          ans.push_back(ans.back()^i);
        }

        return ans;
    }
};